static/js/lib/pearson.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+/* global SETTINGS:false */
+import R from 'ramda';
+
+export const staticFormEntries: Array<[string, string]> = [
+  ["ACTION", "scheduleExam"],
+  ["CLIENT_CODE", SETTINGS.EXAMS_SSO_CLIENT_CODE],
+];
+
+const ssoFormEntries = (studentId, timestamp, timeout, ssoDigest, ssoRedirectURL) => {
+  let baseURL = ssoRedirectURL.replace(/\/$/, '');
+  return staticFormEntries.concat([
+    ["EXTERNAL_ERROR_URL", `${baseURL}/pearson/error`],
+    ["EXTERNAL_LOGOUT_URL", `${baseURL}/pearson/logout`],
+    ["EXTERNAL_RETURN_URL", `${baseURL}/pearson/success`],
+    ["EXTERNAL_TIMEOUT_URL", `${baseURL}/pearson/timeout`],
+    ["CLIENT_CANDIDATE_ID", String(studentId)],
+    ["EXTERNAL_PAGE_TIMESTAMP", String(timestamp)],
+    ["EXTERNAL_SESSION_TIMEOUT", String(timeout)],
+    ["EXTERNAL_AUTH_HASH", ssoDigest],
+  ]);
+};
+
+export const createFormInput = R.curry((form, [name, value]) => {
+  let node = document.createElement("input");
+  node.type = "hidden";
+  node.name = name;
+  node.value = value;
+  form.appendChild(node);
+});
+
+const createForm = () => {
+  let form = document.createElement('form');
+  document.body.appendChild(form);
+  form.action = SETTINGS.EXAMS_SSO_URL;
```

Comment:
What do we do if this is not set? Because then the action url ends up being `/dashboard/null`?